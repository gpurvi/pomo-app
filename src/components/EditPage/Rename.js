import React from 'react';
import {withModal} from "../common/Modal";
import {Button, ModalHeader, ModalBody, ModalFooter, Input} from 'reactstrap';


export class Rename extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            value: this.props.initValue
        };

        this.ref = React.createRef();

        this.onChangeHandler = this.onChangeHandler.bind(this);
        this.onClickHandler = this.onClickHandler.bind(this);
    }

    //this is hack to get input always focused
    componentDidUpdate() {
        this.ref.current.focus();
    }

    onChangeHandler(e) {
        const value = e.target.value;
        this.setState(() => ({value}));
    }

    async onClickHandler(e) {
        e.preventDefault();
        this.props.renameOnClick(this.state.value, this.props.id);
        this.props.toggle();
    }

    render() {
        const {modalName} = this.props;
        return (
            <React.Fragment>
                <ModalHeader toggle={this.props.toggle}>{modalName}</ModalHeader>
                <ModalBody>
                    <Input
                        innerRef={this.ref}
                        type="text"
                        onChange={this.onChangeHandler}
                        value={this.state.value}
                    />
                </ModalBody>
                <ModalFooter>
                    <Button color="primary" onClick={this.onClickHandler}>Save</Button>
                    <Button color="secondary" onClick={this.props.toggle}>Cancel</Button>
                </ModalFooter>
            </React.Fragment>

        );
    }
}

export default withModal(Rename);

