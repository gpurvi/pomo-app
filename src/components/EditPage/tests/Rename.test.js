import React from 'react';
import {shallow} from 'enzyme';
import {Rename} from '../Rename';

jest.mock('./../../common/apiCalls');

it('renders Rename', () => {
    // const wrapper = shallow(<Rename  initValue=''/>);
    // expect(wrapper).toMatchSnapshot();
});
//
// describe('onClickHandler',  () => {
//     it('calls getSessions', async() => {
//         const getSessions = jest.fn();
//         const closeModal = jest.fn();
//         const wrapper = shallow(<Rename
//             closeModal={closeModal}
//             getSessions={getSessions}
//             id={1}/>);
//         await wrapper.find('button').first().simulate('click', {
//             preventDefault() {
//             }
//         });
//         expect(getSessions).toHaveBeenCalled();
//         expect(closeModal).toHaveBeenCalled();
//     });
//
//     // when implement error handling then implement tests
//     // it('calls getSessions', async() => {
//     //     const getSessions = jest.fn();
//     //     const closeModal = jest.fn();
//     //     const wrapper = shallow(<Delete
//     //         closeModal={closeModal}
//     //         getSessions={getSessions} id={1}/>);
//     //     await wrapper.find('button').first().simulate('click', {
//     //         preventDefault() {
//     //         }
//     //     });
//     //     // await wrapper.instance().onClickHandler({preventDefault(){}});
//     //     expect(getSessions).toHaveBeenCalled();
//     // });
// });