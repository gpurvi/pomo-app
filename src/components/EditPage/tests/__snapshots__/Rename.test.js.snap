// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Rename 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Rename
    initValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Rename
</h3>,
        <input
          autoFocus={true}
          onChange={[Function]}
          type="text"
          value=""
/>,
        <button
          onClick={[Function]}
>
          Save
</button>,
        <button
          onClick={undefined}
>
          Cancel
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Rename",
        },
        "ref": null,
        "rendered": "Rename",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoFocus": true,
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Save",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Save",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cancel",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Cancel",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Rename
</h3>,
          <input
            autoFocus={true}
            onChange={[Function]}
            type="text"
            value=""
/>,
          <button
            onClick={[Function]}
>
            Save
</button>,
          <button
            onClick={undefined}
>
            Cancel
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rename",
          },
          "ref": null,
          "rendered": "Rename",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "autoFocus": true,
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Save",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Save",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
